server/src/main/java/org/elasticsearch/index/fieldvisitor/StoredFieldLoader.java

```diff
@@ -55,6 +57,26 @@ public static StoredFieldLoader fromSpec(StoredFieldsSpec spec) {
         return create(spec.requiresSource(), spec.requiredStoredFields());
     }
 
+    /**
+     * Crates a new StoredFieldLaoader using a BlockLoader.FieldsSpec
+     */
+    public static StoredFieldLoader fromSpec(BlockLoader.FieldsSpec spec, boolean forceSequentialReader) {
```

Comment:
Can we add a unit test that tests whether the ignored source field loader is loaded when expected? Maybe we can add a test to `BlockSourceReaderTests` and then also test the produced block? I think current tests in `BlockSourceReaderTests` don't test ignored source?

Reply:
Or maybe adding a test to `IgnoredSourceFieldMapperTests`? But I don't see any block related tests there.